flixel/FlxG.hx

```diff
@@ -379,7 +379,7 @@ class FlxG
 	 */
 	public static inline function resetState():Void
 	{
-		switchState(Type.createInstance(Type.getClass(state), []));
+		switchState(() -> Type.createInstance(Type.getClass(state), []));
```

Comment:
once we start storing the state creator function we can just call that to create a new one, rather than needing to use Type.createInstance.

with Type.createInstance FlxStates can't have constructor args, by passing in functions we can do things like
```hx
FlxG.switchState(()->{ return new MyState(arg1, arg2); });
```
and we'll know how to switch to that state AND reset it

Reply:
Oh, so like a function? I like it, I like it